### PR TITLE
Add basic authentication for ES

### PR DESCRIPTION
**Pull request by CharlesJUDITH**
Right now there is no authentication to ES, this PR will add the basic
authentication. This is not perfect but it's better than nothing.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -147,6 +157,7 @@ func handler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	req, err := http.NewRequest("POST", url, bytes.NewBuffer(b))
+	    req.Header.Add("Authorization","Basic " + basicAuth(esUsername,esPassword))
```
Comment (iksaif) [inline main.go]: indentation looks wrong here, can you run gofmt on the file ?

Comment (iksaif) [reply main.go]: Also only add this header if esUsername is specified

File: main.go [hunk 2]
```diff
@@ -147,6 +157,9 @@ func handler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	req, err := http.NewRequest("POST", url, bytes.NewBuffer(b))
+	if esUsername != "" {
+		req.Header.Add("Authorization", "Basic "+basicAuth(esUsername, esPassword))
```
Comment (iksaif) [inline main.go]: Won't this crash if esPassword isn't specified ? (can you test ?)

Comment (CharlesJUDITH) [reply main.go]: I added one check to verify that part. Of course it's working if you don't want authentication.

Comment (iksaif) [reply main.go]: Apparently it doesn't crash because the default for a string is `""` and not `nil`.
Please revert to this version, ot was better.

**Review by iksaif — APPROVED** [review]

**Pull request merged**